### PR TITLE
Bunch of minor updates to decl-scan module.

### PR DESCRIPTION
**Pull request by czipperz**

### CODE REVIEW

File: haskell-decl-scan.el [hunk 1]
```diff
@@ -217,10 +218,9 @@ only for `imenu' support.")
   (concat literate-haskell-ds-line-prefix haskell-ds-start-decl-re)
   "The regexp that starts a Bird-style literate Haskell declaration.")
 
-(defun haskell-ds-whitespace-p (char)
-  "Test if CHAR is a whitespace character."
-  ;; the nil is a bob/eob test
-  (member char '(nil ?\t ?\n ?\ )))
+(defun haskell-ds-whitespace-p ()
+  "Test if PT is at a whitespace character."
+  (equal (string-to-syntax " ") (syntax-after (point))))
```
Comment (gracjan) [inline haskell-decl-scan.el]: You had `nil` here before to handle `eobp`. Is this needed or not?

File: haskell-decl-scan.el [hunk 2]
```diff
@@ -356,20 +356,15 @@ there."
...
   ;; ensure result is `t' or `nil' instead of just truthy
   (if (or
        ;; is cursor on whitespace
-       (haskell-ds-whitespace-p (following-char))
+       (haskell-ds-whitespace-p)
        ;; http://emacs.stackexchange.com/questions/14269/how-to-detect-if-the-point-is-within-a-comment-area
        ;; is cursor at begging, inside, or end of comment
-       (let ((fontfaces (get-text-property (or pt
-                                               (point)) 'face)))
+       (let ((fontfaces (get-text-property (point) 'face)))
```
Comment (gracjan) [inline haskell-decl-scan.el]: I can't add comment on a line that follows, so I'm putting it here: comments in haskell have structure and we use more faces than just `font-lock-comment-face`: `haskell-pragma-face`, `haskell-liquid-haskell-annotation-face`, `haskell-literate-comment-face`. Sometimes we also add `(:weight bold)` and `(:slant italic)`.

File: haskell-decl-scan.el [hunk 3]
```diff
@@ -382,18 +377,17 @@ there."
...
 (defun haskell-ds-line-commented-p ()
-  "Tests if all characters from `point' to `end-of-line' pass
-`haskell-ds-comment-p'"
-  (let ((r t))
-    (while (and r (not (eolp)))
-      (if (not (haskell-ds-comment-p))
-          (setq r nil))
-      (forward-char))
-    r))
+  "Test if all characters from `point' to `end-of-line' pass `haskell-ds-comment-p'."
```
Comment (gracjan) [inline haskell-decl-scan.el]: This function is horrible. The way to find end of current comment is to use `(parse-partial-sexp (point) (point-max) nil nil (syntax-ppss) 'syntax-table)`. Examples are in `haskell-font-lock.el`.

**Comment by gracjan** [comment]
Do you want me to merge as is and fix the functionality later? Or fix the functionality in place while we are at it?

**Comment by czipperz** [comment]
I'll fix it tonight.

File: haskell-decl-scan.el (hunk 1 above)
Comment (czipperz) [inline haskell-decl-scan.el]: It needs to have eolp or'd to it.

File: haskell-decl-scan.el (hunk 3 above)
Comment (czipperz) [inline haskell-decl-scan.el]: The only usage of this function is to go back to the end of a definition.  That requires going through all lines that are composed only of white space and comments.

File: haskell-decl-scan.el (hunk 2 above)
Comment (czipperz) [inline haskell-decl-scan.el]: The weight and slant already work.  I can add these other faces to the member test if you want.

Comment (gracjan) [inline haskell-decl-scan.el]: Yes, they have to be there. Otherwise declscan will not work.

File: haskell-decl-scan.el (hunk 3 above)
Comment (gracjan) [inline haskell-decl-scan.el]: Use `(parse-partial-sexp (point) (point-max) nil nil (syntax-ppss) 'syntax-table)` for this purpose.

**Comment by gracjan** [comment]
This new github review workflow is confusing. Anyway, since we are chanigng decl-scan we should make it better in those two places I mentioned.

**Comment by geraldus** [comment]
> This new github review workflow is confusing

The most confusing for me is __Comment__ option of review.  I supposed that it should simply add comment to review, but instead it submits review as simple comments :D

**Comment by czipperz** [comment]
I'm going to close this pull, fix it up and reopen it. Thanks for the suggestions @gracjan

File: haskell-decl-scan.el (hunk 2 above)
Comment (czipperz) [inline haskell-decl-scan.el]: Thanks for the help.

**Comment by gracjan** [comment]
You can reopen and force-push new set of commits. Github is pretty good at sorting out what has happened.

**Comment by czipperz** [comment]
That's the plan

**Pull request closed without merging**